clients/admin-ui/src/features/integrations/useIntegrationFilterTabs.tsx

```diff
@@ -0,0 +1,41 @@
+import { useState } from "react";
+
+import { IntegrationTypeInfo } from "~/features/integrations/add-integration/allIntegrationTypes";
+
+export enum IntegrationFilterTabs {
+  ALL = "All",
+  DATABASE = "Database",
+  DATA_WAREHOUSE = "Data Warehouse",
+}
+
+const useIntegrationFilterTabs = (integrationTypes?: IntegrationTypeInfo[]) => {
+  const tabs = Object.values(IntegrationFilterTabs);
+
+  const [tabIndex, setTabIndex] = useState(0);
+  const currentTab = tabs[tabIndex];
+
+  const [isFiltering, setIsFiltering] = useState(false);
+
+  const onChangeFilter = (newIndex: number) => {
+    setIsFiltering(true);
+    setTabIndex(newIndex);
+    setTimeout(() => setIsFiltering(false), 100);
```

Comment:
Putting in a fake "loading" state for a short period makes the change in contents feel less jarring to me-- not married to this if we're against it.

Reply:
I think small artificial delays are valid if it improves the UX. We can keep it.